beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -163,7 +162,7 @@ func (s *Service) updateFinalized(ctx context.Context, cp *ethpb.Checkpoint) err
 
 	fRoot := bytesutil.ToBytes32(cp.Root)
 	optimistic, err := s.cfg.ForkChoiceStore.IsOptimistic(fRoot)
-	if err != nil && err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
+	if err != nil && err != doublylinkedtree.ErrNilNode {
```

Comment:
errors.Is is the preferable approach to checking errors in this manner.

```suggestion
	if err != nil && !errors.Is(err, doublylinkedtree.ErrNilNode) {
```

https://pkg.go.dev/errors#pkg-overview